ansible/roles/common-cartridge/tasks/main.yml

```diff
@@ -8,14 +8,28 @@
   tags:
     - configmap
 
+- name: External Secret
+  kubernetes.core.k8s:
+    kubeconfig: ~/.kube/config
+    namespace: "{{ NAMESPACE }}"
+    template: external-secret.yml.j2
+    state: "{{ 'present' if 
+               WITH_BRANCH_MONGO_DB_MANAGEMENT is defined and WITH_BRANCH_MONGO_DB_MANAGEMENT|bool and
+               WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool
+               else 'absent'}}"

```

Comment:
Intressting that his works, is your intenntion right to place the condition for `WITH_COMMON_CARTRIDGE` there?

Reply:
It think so. Only if common cartridge is enabled we need the External Secret for it otherwise it can be removed. Do you see a problem there that I am overlooking?

Reply:
It was only the differenc where `WITH_COMMON_CARTRIDGE is defined and WITH_COMMON_CARTRIDGE|bool` is place what me wonder.